.github/actions/yarn-install/action.yml

```diff
@@ -4,16 +4,32 @@ inputs:
     type: string
     required: false
     default: '20.x'
+  recreate-yarn-lock:
+    type: boolean
+    required: false
+    default: false
 
 runs:
   using: "composite"
   steps:
   - uses: actions/setup-node@v4
+    name: Setup Node With Yarn Cache
+    if: ${{ inputs.recreate-yarn-lock != 'true' }}
```

Comment:
Do we need this if condition? Seems like L29 is sufficient.

Reply:
Cache can still be used even if `yarn.lock` is removed.